assets/js/blocks/product-query/variations/elements/add-to-cart-button.tsx

```diff
@@ -0,0 +1,109 @@
+/**
+ * External dependencies
+ */
+import { isFeaturePluginBuild } from '@woocommerce/block-settings';
+import { Icon, button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { addFilter } from '@wordpress/hooks';
+import { select } from '@wordpress/data';
+import { EditorBlock } from '@woocommerce/types';
+import { ElementType } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { registerElementVariation } from './utils';
+
+const BLOCK_TITLE: string = __(
+	'Add to Cart Button',
+	'woo-gutenberg-products-block'
+);
+const BLOCK_ICON: JSX.Element = (
+	<Icon icon={ button } className="wc-block-editor-components-block-icon" />
+);
+const BLOCK_DESCRIPTION: string = __(
+	'Display a call to action button which either adds the product to the cart, or links to the product page.',
+	'woo-gutenberg-products-block'
+);
+
+export const CORE_NAME = 'core/buttons';
+export const VARIATION_NAME = 'woocommerce/product-add-to-cart-button';
+
+if ( isFeaturePluginBuild() ) {
+	registerElementVariation( CORE_NAME, {
+		blockDescription: BLOCK_DESCRIPTION,
+		blockIcon: BLOCK_ICON,
+		blockTitle: BLOCK_TITLE,
+		variationName: VARIATION_NAME,
+	} );
+}
+
+/**
+ * This helps us in modifying the UI in editor
+ * for the button block variation
+ */
+export const withCoreButtonVariation =
+	< T extends EditorBlock< T > >( BlockEdit: ElementType ) =>
+	( props ) => {
+		if ( props.name === 'core/button' ) {
+			/**
+			 * Get the parent block to check if it is a variation
+			 */
+			const coreEditor = select( 'core/block-editor' );
+			const parentBlocks = coreEditor.getBlockParents(
+				props.clientId,
+				true
+			);
+			const parentButtonsBlock = coreEditor.getBlock( parentBlocks[ 0 ] );
+
+			const isWoocommerceVariation =
+				VARIATION_NAME ===
+				parentButtonsBlock?.attributes?.__woocommerceNamespace;
+
+			if ( isWoocommerceVariation && ! props.attributes?.text?.length ) {
+				// Set initial attributes when block is added to the editor.
+				props.setAttributes( {
+					...props.attributes,
+					text: __( 'Add to cart', 'woo-gutenberg-products-block' ),
+					style: ! props.attributes.style && {
+						spacing: {
+							padding: {
+								top: 'var:preset|spacing|30',
+								right: 'var:preset|spacing|40',
+								bottom: 'var:preset|spacing|30',
+								left: 'var:preset|spacing|40',
+							},
+						},
+					},
+					align: 'center',
+					fontSize: 'small',
+				} );
```

Comment:
Instead of setting them dynamically here, you can actually pass the default attributes to the `registerBlockVariation` function. Since you are using the `registerElementVariation` util, you might want to add this as an argument to that function.

Reply:
I am not sure if we can pass attributes in `registerBlockVariation` because we are registering `core/buttons` as a variation, but we have to change attributes for `core/button`, which is children of `core/buttons`. 

I also tried using `innerBlocks` as follows, but it didn't work.
```
registerElementVariation( CORE_NAME, {
	blockDescription: BLOCK_DESCRIPTION,
	blockTitle: BLOCK_TITLE,
	variationName: VARIATION_NAME,
	innerBlocks: [
		[
			'core/button',
			{
				text: __( 'Add to cart', 'woo-gutenberg-products-block' ),
				align: 'center',
			},
			[],
		],
	],
} );
```
You can see the code I tried in [this commit](https://github.com/woocommerce/woocommerce-blocks/commit/6c9069e9cb65abf11271ded9f855f905fc489c39).

Am I missing something? 🤷🏻‍♂️

Reply:
Mm, I don't think you are missing something, this code seems correct to me. What didn't work about passing the attributes to this `innerBlocks` configuration?